components/edit-collective/sections/SetupTwoFactorAuth.js

```diff
@@ -0,0 +1,335 @@
+                    />
+                  </H3>
+                  <Container>
+                    <Formik validate={validate} initialValues={initialValues} onSubmit={this.submit}>
+                      {formik => {
+                        const { values, handleSubmit, errors, touched } = formik;
+
+                        return (
+                          <Form>
+                            <StyledInputField
+                              name="TOTPCode"
+                              htmlFor="TOTPCode"
+                              error={touched.TOTPCode && errors.TOTPCode}
+                              label={intl.formatMessage(messages.inputLabel)}
+                              value={values.TOTPCode}
+                              required
+                              mt={2}
+                              mb={3}
+                            >
+                              {inputProps => (
+                                <Field
+                                  as={StyledInput}
+                                  {...inputProps}
+                                  minWidth={300}
+                                  minHeight={100}
+                                  fontSize={'H1'}
+                                  noNumberArrows
+                                  placeholder="123456"
+                                  autoFocus
+                                />
+                              )}
+                            </StyledInputField>
+
+                            <Flex justifyContent={['center', 'left']} mb={4}>
+                              <StyledButton
+                                fontSize="13px"
+                                minWidth="148px"
+                                minHeight="36px"
+                                buttonStyle="primary"
+                                type="submit"
+                                onSubmit={handleSubmit}
+                                loading={submitting}
+                                disabled={values.TOTPCode.length < 6}
+                              >
+                                <FormattedMessage id="TwoFactorAuth.Setup.Form.VerifyButton" defaultMessage="Verify" />
+                              </StyledButton>
+                            </Flex>
+                          </Form>
+                        );
+                      }}
+                    </Formik>
+                  </Container>
+                </Box>
+              </Container>
+            </Fragment>
+          )}
+        </Flex>
+      </Flex>
+    );
+  }
+}
+
+const TwoFactorAuthFragment = gqlV2/* GraphQL */ `
+  fragment TwoFactorAuthFragment on Individual {
+    hasTwoFactorAuth
+  }
+`;
+
+const addTwoFactorAuthToAccountMutation = graphql(
+  gqlV2/* GraphQL */ `
+    mutation AddTwoFactorAuthToAccount($account: AccountReferenceInput!, $token: String!) {
+      addAccountTwoFactorAuth(account: $account, token: $token) {
+        id
+        ...TwoFactorAuthFragment
```

Comment:
While correct, this could be simplified by just writing. No `TwoFactorAuthFragment` needed.

```
... on Individual {
  hasTwoFactorAuth
}
```

Reply:
This could also be simplified by making `addAccountTwoFactorAuth` returns `Individual` instead of `Account`.
Possibly renaming to: `addTwoFactorAuthTokenToIndividual`

Reply:
Thanks! I had not yet used fragments so was just following the Apollo docs. Good to know if can be written shorter.